vignettes/hys_law.Rmd

```diff
@@ -0,0 +1,197 @@
+
+* 01-701-1015, 01-701-1023, 01-701-1028, 01-701-1033, 01-701-1034, 01-701-1047, 01-701-1097, 01-705-1186, 01-705-1292, 01-705-1310, 01-708-1286. 
+
+```{r, echo = FALSE}
+adlb <- admiral_adlb
+head(adlb) %>%
+  dataset_vignette()
+```
+
+
+## Creating the Appropriate CRITX and CRITXFL Columns {#columns}
+
+A standard convention of ADLBHY datasets, are various `CRITX` and `CRITXFL` columns to describe the conditions necessary to reach that particular criterion of Hy's Law and the actual flag itself to indicate whether or not the condition was reached.
+
+Using `mutate()`, `call_derivation()` and `derive_var_merged_exist_flag()`, we can create these columns that indicate the the 3-fold or greater than upper limit of normal of ALT/AST and the 2-fold or greater than upper limit of normal of BILI.
+
+To increase visibility and for simplicity, we will retain only columns that are relevant to a Hy's Law analysis for now. 
+
+```{r}
+adlb_annotated <- adlb %>%
+  filter(PARAMCD %in% c("AST", "ALT", "BILI") & is.na(DTYPE)) %>%
+  mutate(CRIT1 = case_when(
+    PARAMCD == "AST" ~ "AST >=3xULN",
+    PARAMCD == "ALT" ~ "ALT >=3xULN",
+    PARAMCD == "BILI" ~ "BILI >=2xULN"
+  )) %>%
+  call_derivation(
+    .,
+    derivation = derive_var_merged_exist_flag,
+    dataset_add = .,
+    by_vars = exprs(USUBJID, LBSEQ, PARAMCD, ADT),
+    variable_params = list(
+      params(
+        new_var = CRIT1FL,
+        condition = (
+          (AVAL / ANRHI >= 3 & PARAMCD %in% c("AST", "ALT")) |
+            (AVAL / ANRHI >= 2 & PARAMCD == "BILI")
+        )
+      )
+    )
+  ) %>%
+  select(STUDYID, USUBJID, TRT01A, PARAMCD, LBSEQ, ADT, AVISIT, ADY, AVAL, ANRHI, CRIT1, CRIT1FL)
+```
+
+```{r, echo = FALSE}
+dataset_vignette(adlb_annotated)
+```
+
+
+## Subsetting by LBTESTCD and Joining by Potential Events {#joins}
+
+If an elevated ALT/AST event reaches the threshold for Hy's Law, we need to search for any elevated BILI events within a certain time-window, usually up to 14 days after the elevated ALT/AST event (this window may vary by organization). By,
+
+  1) Splitting our dataset into its ALT/AST and BILI subsets, respectively, and 
+  2) Joining these two datasets using `derive_vars_joined()` while using the `filter_join` argument to only join together the relevant flagged BILI records that have a corresponding flagged ALT/AST record (prior up to 14 days but may vary for trial/organization) that would indicate a potential Hy's Law event, 
+
+the resulting dataset is helpful for deriving additional parameters. The dataset may also prove useful for a listing where you have to display the two lab-records in one row to showcase the potential event.  
+
+```{r}
+altast_records <- adlb_annotated %>%
+  filter(PARAMCD %in% c("AST", "ALT"))
+
+bili_records <- adlb_annotated %>%
+  filter(PARAMCD %in% c("BILI"))
+
+hylaw_records <- derive_vars_joined(
+  dataset = altast_records,
+  dataset_add = bili_records,
+  by_vars = exprs(STUDYID, USUBJID, ADY),
```

Comment:
This is an excellent point, I think I was confused because without it I was getting 
"`get_duplicates_dataset()` to access the duplicate records" warnings but in a way that's the point to cross-check each ALT/AST record with every BILI record for the time-window search.